src/migrations/1631992797354-organization-verification.ts

```diff
@@ -0,0 +1,56 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class organizationVerification1631992797354
+  implements MigrationInterface
+{
+  name = 'organizationVerification1631992797354';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      "CREATE TABLE `OrganizationVerification` (`id` char(36) NOT NULL, `createdDate` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6), `updatedDate` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6), `version` int NOT NULL, `organizationID` varchar(255) NOT NULL, `status` varchar(255) NOT NULL DEFAULT 'not-verified', `authorizationId` char(36) NULL, `lifecycleId` char(36) NULL, UNIQUE INDEX `REL_3795f9dd15ef3ef2dd1d27e309` (`authorizationId`), UNIQUE INDEX `REL_22be0d440df7972d9b3a94aa6d` (`lifecycleId`), PRIMARY KEY (`id`)) ENGINE=InnoDB"
+    );
+    await queryRunner.query(
+      'ALTER TABLE `organization` DROP COLUMN `verificationType`'
+    );
+    await queryRunner.query(
+      'ALTER TABLE `organization` ADD `verificationId` char(36) NULL'
+    );
+    await queryRunner.query(
+      'ALTER TABLE `organization` ADD UNIQUE INDEX `IDX_95bbac07221e98072beafa6173` (`verificationId`)'
+    );
+    await queryRunner.query(
+      'CREATE UNIQUE INDEX `REL_95bbac07221e98072beafa6173` ON `organization` (`verificationId`)'
+    );
+    await queryRunner.query(
+      'ALTER TABLE `OrganizationVerification` ADD CONSTRAINT `FK_3795f9dd15ef3ef2dd1d27e309c` FOREIGN KEY (`authorizationId`) REFERENCES `authorization_policy`(`id`) ON DELETE SET NULL ON UPDATE NO ACTION'
+    );
+    await queryRunner.query(
+      'ALTER TABLE `OrganizationVerification` ADD CONSTRAINT `FK_22be0d440df7972d9b3a94aa6d5` FOREIGN KEY (`lifecycleId`) REFERENCES `lifecycle`(`id`) ON DELETE SET NULL ON UPDATE NO ACTION'
+    );
+    await queryRunner.query(
+      'ALTER TABLE `organization` ADD CONSTRAINT `FK_95bbac07221e98072beafa61732` FOREIGN KEY (`verificationId`) REFERENCES `OrganizationVerification`(`id`) ON DELETE SET NULL ON UPDATE NO ACTION'
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
```

Comment:
That migration doesn't work:
> ts-node -r tsconfig-paths/register ./node_modules/typeorm/cli.js --config src/config/typeorm.cli.config.ts "migration:revert"

query: SELECT * FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA` = 'alkemio' AND `TABLE_NAME` = 'migrations_typeorm'
query: SELECT * FROM `alkemio`.`migrations_typeorm` `migrations_typeorm` ORDER BY `id` DESC
7 migrations are already loaded in the database.
organizationVerification1631992797354 is the last executed migration. It was executed on Sat Sep 18 2021 22:19:57 GMT+0300 (Eastern European Summer Time).
Now reverting it...
query: START TRANSACTION
query: ALTER TABLE `OrganizationVerification` DROP FOREIGN KEY `FK_22be0d440df7972d9b3a94aa6d5`
query: ALTER TABLE `OrganizationVerification` DROP FOREIGN KEY `FK_3795f9dd15ef3ef2dd1d27e309c`
query: ALTER TABLE `organization` DROP COLUMN `verificationId`
query failed: ALTER TABLE `organization` DROP COLUMN `verificationId`
error: Error: ER_FK_COLUMN_CANNOT_DROP: Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'
    at Query.Sequence._packetToError (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:375:28)
    at Socket.emit (domain.js:470:12)
    --------------------
    at Protocol._enqueue (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at PoolConnection.query (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:198:25)
    at MysqlQueryRunner.<anonymous> (/home/valentin/source/repos/alkemio/server/src/driver/mysql/MysqlQueryRunner.ts:184:36)
    at step (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:143:27)
    at Object.next (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:124:57)
    at fulfilled (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:114:62)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  code: 'ER_FK_COLUMN_CANNOT_DROP',
  errno: 1828,
  sqlMessage: "Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'",
  sqlState: 'HY000',
  index: 0,
  sql: 'ALTER TABLE `organization` DROP COLUMN `verificationId`'
}
query: ROLLBACK
Error during migration revert:
QueryFailedError: ER_FK_COLUMN_CANNOT_DROP: Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'
    at QueryFailedError.TypeORMError [as constructor] (/home/valentin/source/repos/alkemio/server/src/error/TypeORMError.ts:7:9)
    at new QueryFailedError (/home/valentin/source/repos/alkemio/server/src/error/QueryFailedError.ts:9:9)
    at Query.<anonymous> (/home/valentin/source/repos/alkemio/server/src/driver/mysql/MysqlQueryRunner.ts:195:37)
    at Query.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:526:10)
    at Query._callback (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:488:16)
    at Query.Sequence.end (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
    at Query.ErrorPacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Query.js:92:8)
    at Protocol._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:43:10) {
  query: 'ALTER TABLE `organization` DROP COLUMN `verificationId`',
  parameters: undefined,
  driverError: Error: ER_FK_COLUMN_CANNOT_DROP: Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'
      at Query.Sequence._packetToError (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
      at Query.ErrorPacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
      at Protocol._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)
      at Parser._parsePacket (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:433:10)
      at Parser.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Parser.js:43:10)
      at Protocol.write (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:38:16)
      at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:88:28)
      at Socket.<anonymous> (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:526:10)
      at Socket.emit (events.js:375:28)
      at Socket.emit (domain.js:470:12)
      --------------------
      at Protocol._enqueue (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/protocol/Protocol.js:144:48)
      at PoolConnection.query (/home/valentin/source/repos/alkemio/server/node_modules/mysql/lib/Connection.js:198:25)
      at MysqlQueryRunner.<anonymous> (/home/valentin/source/repos/alkemio/server/src/driver/mysql/MysqlQueryRunner.ts:184:36)
      at step (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:143:27)
      at Object.next (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:124:57)
      at fulfilled (/home/valentin/source/repos/alkemio/server/node_modules/tslib/tslib.js:114:62)
      at processTicksAndRejections (internal/process/task_queues.js:95:5) {
    code: 'ER_FK_COLUMN_CANNOT_DROP',
    errno: 1828,
    sqlMessage: "Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'",
    sqlState: 'HY000',
    index: 0,
    sql: 'ALTER TABLE `organization` DROP COLUMN `verificationId`'
  },
  code: 'ER_FK_COLUMN_CANNOT_DROP',
  errno: 1828,
  sqlMessage: "Cannot drop column 'verificationId': needed in a foreign key constraint 'FK_95bbac07221e98072beafa61732'",
  sqlState: 'HY000',
  index: 0,
  sql: 'ALTER TABLE `organization` DROP COLUMN `verificationId`'
}